Allow the Administrators group to sync

robdaemon opened the pull request:
Allows the Administrators group to sync from LDAP or SAML providers.
This was previously not allowed to avoid a user accidentally locking
themselves out of the admin group.

Fixes metabase/metabase-enterprise#445

Commit:
Allow the Administrators group to sync

Allows the Administrators group to sync from LDAP or SAML providers.
This was previously not allowed to avoid a user accidentally locking
themselves out of the admin group.

Fixes metabase/metabase-enterprise#445

robdaemon commented:
I'll rebase this once #13632 is merged

dacort commented:
Tested this locally and it worked - one question, though, will this now _remove_ users from the administrators group?

robdaemon commented:
@dacort that is a very good question and not one I know if we have a test for

camsaul commented:
Do we still want to merge this based on discussion in Slack today? Not sure what we decided WRT the behavior we want.

If we *do* merge this, I think we should definitely get a test around removing admins as well. We have to get the behavior here locked down 100% since admin access is pretty much the keys to the kingdom

camsaul reviewed (changes requested):
We need a test for removing someone from the Admin group

Commit:
test that admins are properly removed

Test that when a sync happens, if the user was previously in the administrators group and no longer is, they are properly removed from that group.

robdaemon commented:
@camsaul okay added an explicit test for that

camsaul reviewed (approved):
LGTM
